### DISCUSSION TITLE
Add a separate Y-value check for "despawn-ranges"

### DISCUSSION THREAD

**Discussion by CatTeaA** [PaperMC/Paper#10520] [Ideas]
### Is your feature request related to a problem?

When the number of players reaches dozens of people, the high simulation distance cannot be afforded. My simulation distance is 5.
And then Monsters outside of simulation distance do not disappear randomly.
The <spigot.yml> file also has an option:
[mob-spawn-range: (number)](https://docs.papermc.io/paper/reference/spigot-configuration#world_settings_default_mob_spawn_range)
This option means the range, in chunks, from the player that mobs can spawn.
--- If a player is AFK (Away From Keyboard), as time passes, monsters will still move beyond the simulation distance and will not vanish, resulting in an accumulation of a large number of monsters

Peper has an option for this[<Despawn Range (hard)>](https://docs.papermc.io/paper/reference/world-configuration#entities_spawning_despawn_ranges_monster_hard)
Hopefully it would be nice to separate the Y-value checks separately.
For example, monsters can be configured to disappear outside of 128 from the player's Y value.The X and Z values disappear beyond 80.
(It is guaranteed that there is no need to modify the afk spots of the biological farm)

If this feature request can be implemented, it would be very beneficial for servers with a high player count, allowing them to maintain the 'vanilla' experience of the server as much as possible.

![image](https://github.com/PaperMC/Paper/assets/135555687/b45dc5b5-3d9d-4a0e-8fcf-c19cbe14461b)

This request is a bit like "[tracking-range-y](https://docs.papermc.io/paper/reference/world-configuration#entities_tracking_range_y_enabled)".
More granular configuration options will help a lot of people.
Hopefully someone will be able to create a PR for this.

### Describe the solution you'd like.

Add a separate Y-value check for "despawn-ranges"
despawn-ranges-y

### Describe alternatives you've considered.

https://hangar.papermc.io/Kyle/harderdespawn
I'm using this plugin right now.but only when the chunk is unloaded, the mob disappears. Cannot fully resolve this issue

If a player is AFK (Away From Keyboard), as time passes, monsters will still move beyond the simulation distance and will not vanish, resulting in an accumulation of a large number of monsters.

(It is even abused by players, and when the player encounters a monster fight, he quits the game and rejoins, and the monsters are all gone).
It would be nice if "despawn-ranges" could be configured with a Y value separately.
"despawn-ranges-y"

### Other

https://github.com/PaperMC/Paper/issues/9476
Thanks to electronicboy suggestion, I closed the previous Feature Request.
The previous Feature Request was a bit weird

**Comment by CatTeaA** [comment]
When this PR is merged, it will perfectly solve the issue I'm currently facing, I'm very much looking forward to it.
https://github.com/PaperMC/Paper/pull/10440

**Comment by lynxplay** [comment]
Resolved :)
